Implemented redux store with auth slice

Vinyl-Davyl opened the pull request.

williamssam commented:
Hi @Vinyl-Davyl , Why did you switch off "noImplicitAny"??

pablo-clueless commented on client/src/store/slices/auth.ts:
```diff
@@ -0,0 +1,52 @@
+        logout: (state) => {
+            state.user = null
+            state.isLoggedIn = false
+            localStorage.removeItem('team-menber')
```
`localStorage` key should match with what we have on line 36

Vinyl-Davyl replied:
Yeah, Resolved typo

pablo-clueless commented on client/src/types/index.ts:
```diff
@@ -0,0 +1,18 @@
+
```
Use the schema here, https://dbdiagram.io/d/63c79244296d97641d7a6111, to create the user type

Vinyl-Davyl replied:
Okay

Vinyl-Davyl commented:
> Hi @Vinyl-Davyl , Why did you switch off "noImplicitAny"?? and you also removed ErrorBoundary component from app.tsx??

Error boundary component?? Not sure about that. Didn't change any file there. But I switched off non Implicit any and some other stuff to resolve some type warning. Can always take the back though and find a way around
